core-util/CriticalSectionLock.h

```diff
@@ -51,7 +52,24 @@ class CriticalSectionLock {
 public:
     CriticalSectionLock() {
 #ifdef TARGET_NORDIC
-        sd_nvic_critical_region_enter(&_state);
+        // get the state of exceptions for the CPU
+        _PRIMASK_state = __get_PRIMASK();
+
+        // if exceptions are not enabled, there is nothing more to do
+        if(_PRIMASK_state == 1) {
+          _use_softdevice_routine = false;
```

Comment:
No, as long as a CriticalSectionLock is declared on the stack. In the current form, recursive locks are already allowed.